client/src/components/ApplicantProfile.js

```diff
@@ -23,23 +23,38 @@ const ApplicantProfile = () => {
   ]);
 
   const [profileCreated, setProfileCreated] = useState(null);
+  const addMoreExperience = () => {
+    setExperienceData([
+      ...experienceData,
+      {
+        company: "",
+        job_title: "",
+        description: "",
+        start_date: "",
+        end_date: ""
+      }
+    ]);
+  };
 
   const handleChange = (event, result) => {
     const { name, value } = result || event.target;
     setProfileData({ ...profileData, [name]: value });
   };
-
-  const handleExperience = ({ event, targetData, experienceIndex }) => {
+  const updateExperienceData = (value, name, index) =>
+    experienceData.map((experience, experienceIndex) => {
```

Comment:
I think this could be more readably (very arguable and subjective) written as something like

updateExperiences = (value, name, index) => {
  const updatedExperienceData = [...experiencesData];
  updatedExperiences[index] = {...updatedExperienceData[index], [name] : value};
  return updatedExperienceData;
}